### PR TITLE
Fix issue of global representation

### PR DESCRIPTION
**Pull request by LeoneChen**
https://github.com/SVF-tools/SVF/issues/1483

This patch is not widely tested, and we need to think more about how to fix the root causes

### CODE REVIEW

File: svf-llvm/include/SVF-LLVM/SVFIRBuilder.h [hunk 1]
```diff
@@ -97,6 +97,7 @@ class SVFIRBuilder: public llvm::InstVisitor<SVFIRBuilder>
     /// GetObject - Return the object node (stack/global/heap/function) according to a LLVM Value
     inline NodeID getObjectNode(const Value* V)
     {
+        V = LLVMUtil::getGlobalRep(V);
```
Comment (yuleisui) [inline svf-llvm/include/SVF-LLVM/SVFIRBuilder.h:100]: Global rep should always be a global. The getObject node can return a null global obj. Why adding this line?

Comment (LeoneChen) [reply svf-llvm/include/SVF-LLVM/SVFIRBuilder.h:100]: The reason why I add `V = LLVMUtil::getGlobalRep(V);` is that `val` is converted to global representation in `collectObj` to perform `insert` on `symInfo->objSymMap`
https://github.com/SVF-tools/SVF/blob/0eae65388008dbfafc0287873bfb4c8819582183/svf-llvm/lib/SymbolTableBuilder.cpp#L318-L330

Comment (LeoneChen) [reply svf-llvm/include/SVF-LLVM/SVFIRBuilder.h:100]: Let me give you a case

Comment (LeoneChen) [reply svf-llvm/include/SVF-LLVM/SVFIRBuilder.h:100]: Stack is like this (pictures below may be more vivid), `objSymMap` is updated in above `collectObj`
image
image
image

Comment (LeoneChen) [reply svf-llvm/include/SVF-LLVM/SVFIRBuilder.h:100]: But I'm not sure whether this fix is accurate or not

File: svf-llvm/include/SVF-LLVM/SVFIRBuilder.h [hunk 2]
```diff
@@ -97,6 +97,7 @@ class SVFIRBuilder: public llvm::InstVisitor<SVFIRBuilder>
...
         SVFValue* svfVal = LLVMModuleSet::getLLVMModuleSet()->getSVFValue(V);
```
Comment (yuleisui) [inline svf-llvm/include/SVF-LLVM/SVFIRBuilder.h:101]: I think it is good to put your ` V = LLVMUtil::getGlobalRep(V);` into method `getSVFValue `

**Comment by LeoneChen** [comment]
Since it influences all usage about global value, does it will cause new problem?

**Comment by yuleisui** [comment]
> Since it influences all usage about global value, does it will cause new problem?

It looks to be a clean fix for global. Would you test more and larger bcs before I merge this?

**Comment by LeoneChen** [comment]
I agree that more testing is needed and I will report here if I find something new. Does SVF have some unit test cases and integration test cases that can perform the testing?

**Comment by yuleisui** [comment]
> I agree that more testing is needed and I will report here if I find something new. Does SVF have some unit test cases and integration test cases that can perform the testing?

Your patch has already passed SVF tests in our CI. Would be good for you to test some large bcs and report back so I can merge your pull request

**Comment by LeoneChen** [comment]
ok

**Pull request merged**